monai/metrics/meandice.py

```diff
@@ -44,6 +43,11 @@ class DiceMetric(CumulativeIterationMetric):
         ignore_empty: whether to ignore empty ground truth cases during calculation.
             If `True`, NaN value will be set for empty ground truth cases.
             If `False`, 1 will be set if the predictions of empty ground truth cases are also empty.
+        compute_sample: if ``True``, each sample's dice metric will be computed before reduction.
```

Comment:
Maybe call it `sample_wise`? @wyli What do you think?

Thanks.

Reply:
perhaps we just modify the existing `reduction` to support this sample mode? also for the typo, `dice` should be changed to `Dice`

Reply:
I agree with @wyli on just modifying `reduction` method and having something like `reduction_sample` and use the `filename` attribute to accumulate the metrics.